libs/base/Data/String.idr

```diff
@@ -132,7 +133,8 @@ unlines' (l::ls) = l ++ '\n' :: unlines' ls
 ||| ```
 export
 unlines : List String -> String
-unlines = pack . unlines' . map unpack
+unlines [""] = "\n"
+unlines = join "\n"
```

Comment:
All patterns in pattern matching must have the same count of arguments they are matching at, so you'd better to have `unlines ss = join "\n" ss` for the second line.

Anyway, this implementation is not equivalent to the current `unlines` implementation. For example, `unlines ["", ""]` now gives `"\n\n"`, your implementation gives `"\n"`.

Reply:
Just to further the point made in the above comment, here's a reference to when the semantics of lines/unlines were most recently changed: https://github.com/idris-lang/Idris2/pull/1585

That is, the current semantics were intentional instead of accidental so we should honor them in the changes made in this PR. That might make the suggestion of using `join` less obvious of a win, at least for unlines, though I wish I'd thought to mention that on the open GitHub ticket in the first place.

Reply:
Should unlines be written to use `join`? I ended up writing the following which doesn't include join so I'm considering reverting this specific change since it won't be any different to the original code.

https://github.com/idris-lang/Idris2/blob/54332752f0cb43645f365d6780a00c3348a4377e/libs/base/Data/String.idr#L147-L149

Reply:
Since the function isn't publicly exported and it is transformed to `fastUnlines` at runtime, I suppose it's neither here nor there whether it unpacks and runs on characters or not. I think I slightly prefer your new approach personally.